lib/knapsack_pro/config/temp_files.rb

```diff
@@ -0,0 +1,46 @@
+module KnapsackPro
+  module Config
+    class TempFiles
+      def self.temp_directory_path
+        temp_files = new
+        temp_files.ensure_temp_directory_exists
+        temp_files.temp_directory_path
+      end
+
+      def ensure_temp_directory_exists
+        unless File.exists?(gitignore_file_path)
+          create_temp_directory
+          create_gitignore_file
+        end
+      end
+
+      # relative to the directory where you run knapsack_pro gem (user's project)
+      def temp_directory_path
+        '.knapsack_pro'
+      end
+
+      private
+
+      def create_temp_directory
+        FileUtils.mkdir_p(temp_directory_path)
+      end
+
+      def gitignore_file_path
+        File.join(temp_directory_path, '.gitignore')
```

Comment:
> Do I understand correctly, that we'll create our own .gitignore file in our new temp directory?

Yes.

> Does it mean that there would now need to be the .knapsack_pro directory commited in the project (with just the .gitignore file committed)?

No. Because we also ignore `.gitignore` file with `*` char in `.gitignore` file.

`.gitignore` file is created during gem runtime. It does not have to be committed into repo even when you run knapsack pro locally in development because all content of `.knapsack_pro` is ignored so it means for git the `.knapsack_pro` directory is empty. It's not possible to commit an empty directory in git so this way `.knapsack_pro` with the `.gitignore` file won't be committed.

Reply:
Cool, I hoped it works the way you described it. Great to know it does indeed! :green_heart: